### PR TITLE
feat[Feature]: create leave request feature

### PR DESCRIPTION
**Pull request by TienTranSi**
UI create leave request of requester
image

### CODE REVIEW

File: src/com/axonactive/mytimeapplication/service/Impl/LeaveRequestServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+			}
+			return false;
+		}).toList();
+	}
+
+	private List<LeaveRequest> getAllLeaveRequestOfEmployee(Employee employee) {
+		if (employee == null) {
+			return new ArrayList<LeaveRequest>();
+		}
+		return leaveRequestDAO.findAll().stream().filter(l -> l.getRequester().equals(employee)).toList();
```
[the remark this answers is not in the document]
Comment (TienTranSi) [reply src/com/axonactive/mytimeapplication/service/Impl/LeaveRequestServiceImpl.java:107]: I will improve it through this task [https://trello.com/c/7pMKbULf/34-retrieve-data-using-criteria-queries](Task34) later, please ignore it for now

File: src/com/axonactive/mytimeapplication/service/Impl/LeaveRequestServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+							.getCms(List.of(leaveRequest.getStartTime(), leaveRequest.getEndTime())),
+					leaveRequest.getStartTime(), leaveRequest.getEndTime());
+		}
+		// checking overlap to do ...
+		return leaveRequestDAO.save(leaveRequest);
+	}
+
+	@Override
+	public LeaveRequest getPrepareLeaveRequestApprovalsOfCurrentUser() {
+		LeaveRequest lr = new LeaveRequest();
```
[the remark this answers is not in the document]
Comment (TienTranSi) [reply src/com/axonactive/mytimeapplication/service/Impl/LeaveRequestServiceImpl.java:125]: Thanks, I will fix it right away

File: src/com/axonactive/mytimeapplication/service/Impl/LeaveRequestServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		}
+	}
+
+	private List<LeaveRequest> getAllLeaveRequestByHOD(Employee headOfDepartment) {
+		if (headOfDepartment == null) {
+			return new ArrayList<LeaveRequest>();
+		}
+		return leaveRequestDAO.findAll().stream().filter(l -> {
+			if (l.getRequester().equals(headOfDepartment))
+				return true;
```
[the remark this answers is not in the document]
Comment (TienTranSi) [reply src/com/axonactive/mytimeapplication/service/Impl/LeaveRequestServiceImpl.java:77]: Sorry about that, I will keep in mind this time

File: src/com/axonactive/mytimeapplication/service/Impl/EmployeeServiceIplm.java [hunk 4]
```diff
@@ -133,7 +132,27 @@ public Employee deleteEmployee(long employeeId)
...
 	public Employee getCurrentUserEmployee() {
+		Ivy.log().debug(ivyService.getSessionUsername());
 		return employeeDAO.getEmployeeByUsername(ivyService.getSessionUsername()).get();
 	}
 
+	@Override
+	public IvyRole getRoleOfCurrentUserEmployee() {
+		List<IvyRole> roles = ivyService.getRoleByUsername(ivyService.getSessionUsername());
+		if (roles.contains(IvyRole.OFFICE_MANAGER))
+			return IvyRole.OFFICE_MANAGER;
```
Comment (natuan2) [inline src/com/axonactive/mytimeapplication/service/Impl/EmployeeServiceIplm.java:143]: Missing curly braces

File: src_hd/com/axonactive/mytimeapplication/ManageEmployee/ManageEmployee.xhtml [hunk 5]
```diff
@@ -249,47 +239,130 @@
...
+							</f:facet>
+							<f:facet name="filter">
+								<p:selectOneMenu id="activeFilter" filter="true"
+									onchange="PF('employeesValue').filter()" style="width: 100%;">
+									<f:selectItem itemLabel="All" itemValue="" />
+									<f:selectItem itemLabel="Yes" itemValue="true" />
+									<f:selectItem itemLabel="No" itemValue="false" />
+								</p:selectOneMenu>
+							</f:facet>
+							<h:outputText value="#{employee.active ? 'Yes' : 'No'}" />
```
Comment (natuan2) [inline src_hd/com/axonactive/mytimeapplication/ManageEmployee/ManageEmployee.xhtml:357]: `Yes` `No` should be from CMS

**Review by natuan2 — APPROVED** [review]

**Pull request merged**